dir.c

```diff
@@ -746,20 +750,20 @@ file_exists_p (const char *name)
   const char *dirend;
   const char *dirname;
   const char *slash;
-#ifdef _GUARDIAN_TARGET
-  FILE *fp;
-#endif
 
 #ifndef NO_ARCHIVES
   if (ar_name (name))
     return ar_member_date (name) != (time_t) -1;
 #endif
 
 #ifdef _GUARDIAN_TARGET
-  fp = fopen(name, "r");
-  fclose(fp);
-  return fp ? 1 : 0;
-#else
+  short error = FILE_GETINFOBYNAME_(name, (short) strlen(name));
+  if (error == 0)
+    return 1;
+  if (error != 13)
+    return 0;
+  // Fall through to standard POSIX parsing.
```

Comment:
Which does use `access()`? Can you point me at where that happens?

Reply:
The code uses `dir_` functions native to GNU Make further down at

https://github.com/ituglib/gmake/blob/ddece16e0155e6eb0b16b4d2202f986886c07eeb/dir.c#L808